modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicStore.java

```diff
@@ -0,0 +1,484 @@
+public class VectorMosaicStore extends ContentDataStore {
+    static final Logger LOGGER = Logging.getLogger(VectorMosaicStore.class);
+
+    @SuppressWarnings("PMD.UnusedPrivateField") // field is used in constructor
+    private final Repository repository;
+
+    DataStore delegateDataStore;
+    boolean tolerant;
+
+    public static final String PREFERRED_SPI_DEFAULT =
+            "org.geotools.data.shapefile.ShapefileDataStoreFactory";
+
+    String preferredSPI = PREFERRED_SPI_DEFAULT;
+
+    public static final String CONNECTION_PARAMETER_KEY_URL = "url";
+
+    public static final String CONNECTION_PARAMETER_KEY_FILE = "file";
+
+    String connectionParameterKey = CONNECTION_PARAMETER_KEY_URL;
+
+    private static final Map<String, String> REQUIRED_VECTOR_MOSAIC_FIELDS =
+            new HashMap<String, String>() {
+                {
+                    put(VectorMosaicGranule.TYPE_NAME_DELEGATE_FIELD, String.class.getName());
+                    put(
+                            VectorMosaicGranule.CONNECTION_PARAMETERS_DELEGATE_FIELD_DEFAULT,
+                            String.class.getName());
+                    put(VectorMosaicGranule.GRANULE_TYPE_NAME, String.class.getName());
+                }
+            };
+    private List<String> delegateTypes;
+    Map<String, VectorMosaicTypeConfiguration> typeMap = new LinkedHashMap<>();
+
+    private static SoftValueHashMap<String, VectorMosaicGranule> GRANULE_STORE_CACHE =
+            new SoftValueHashMap<>();
+
+    static {
+        GeoTools.addChangeListener(
+                new ChangeListener() {
+                    @Override
+                    public void stateChanged(ChangeEvent e) {
+                        synchronized (VectorMosaicStore.class) {
+                            GRANULE_STORE_CACHE.clear();
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Creates a new instance of VectorMosaicStore
+     *
+     * @param delegateStoreName the name of the delegate store. Delegate store must be registered in
+     *     the repository and must point to vector featuretypes with identical attributes.
+     * @param repository the repository to use for lookup of the delegate store.
+     * @param preferredDataStoreSPI the preferred SPI to use for the delegate store.
+     * @param connectionParameterKey the key to use for the connection parameter in the delegate
+     *     store.
+     */
+    public VectorMosaicStore(
+            String delegateStoreName,
+            Repository repository,
+            String preferredDataStoreSPI,
+            String connectionParameterKey) {
+        this.repository = repository;
+
+        this.preferredSPI = preferredDataStoreSPI;
+
+        if (connectionParameterKey != null) {
+            this.connectionParameterKey = connectionParameterKey;
+        }
+        Name name = buildName(delegateStoreName);
+        delegateDataStore = repository.dataStore(name);
```

Comment:
I think the repository would have worked also with simple strings, did you try it out in GeoServer?